src/services/expenses.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+let expenses = [];
+
+const filterExpenses = ({
+  userId,
+  categories,
+  from,
+  to,
+}) => {
+  let filteredExpenses = expenses;
+
+  if (userId) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      expense.userId.toString() === userId
+    ));
+  }
+
+  if (categories) {
+    filteredExpenses = filteredExpenses.filter(expense => (
+      categories.includes(expense.category)
+    ));
+  }
+
+  if (from && to) {
+    filteredExpenses = filteredExpenses.filter(expense => {
+      const expenseDate = new Date(expense.spentAt);
+      const fromDate = new Date(from);
+      const toDate = new Date(to);
```

Comment:
you create it each iteration, move it from the filter
```suggestion
```

Reply:
Done